demo/src/screens/componentScreens/WithScrollReachedScreen/FadedScrollView.tsx

```diff
@@ -0,0 +1,96 @@
+import _ from 'lodash';
+import React, {memo, useCallback} from 'react';
+import {
+  ScrollView,
+  StyleSheet,
+  NativeSyntheticEvent,
+  NativeScrollEvent
+} from 'react-native';
+import {
+  Colors,
+  Text,
+  View,
+  Image,
+  withScrollReached,
+  WithScrollReachedProps
+} from 'react-native-ui-lib';
+
+const FADE_OUT_HEIGHT = 100;
+const fadeImage = require('../../../assets/images/FadeOut.png');
+const WithScrollReached = (props: WithScrollReachedProps) => {
+  const renderItem = useCallback((index: number) => {
+    return (
+      <View key={index} style={styles.item}>
+        <Text>{index + 1}</Text>
+      </View>
+    );
+  }, []);
+
+  const onScroll = useCallback(
+    (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+      _.invoke(props, 'onScroll', event);
+      _.invoke(props, 'scrollReachedProps.onScroll', event);
+    },
+    [props.onScroll, props.scrollReachedProps.onScroll]
+  );
+
+  const renderFade = useCallback(() => {
+    if (
+      (_.isUndefined(props.scrollEnabled) || props.scrollEnabled) &&
+      !props.scrollReachedProps.isScrollAtEnd
+    ) {
+      return (
+        <Image
+          style={styles.fadeOutImage}
+          testID={`${props.testID}.fadeOut`}
+          source={fadeImage}
+        />
+      );
+    }
+  }, [
+    props.scrollEnabled,
+    props.scrollReachedProps.isScrollAtEnd,
+    props.testID
```

Comment:
why do we depend on  `props.scrollEnabled` (I think it's a left over of copy-paste)
And also `props.testID` how does it affect the render in any way?

Reply:
`testID` is a mistake, `scrollEnabled` will technically be needed in a real example.
When I think about it, maybe a better solution for the screen(s) is having a single screen for both HOCs, although it will make it more complicated, it will be a "real" example.
What do you think?

Reply:
For now keep the example screens separated. 
Specifically here, I don't there there's a `props.scrollEnabled` so it's redundant...

Reply:
Done